### PR TITLE
1489 comments module

### PR DESCRIPTION
**Pull request by ledsoft**
https://kbss.felk.cvut.cz/redmine/issues/1489

### CODE REVIEW

File: src/action/__tests__/AsyncCommentActions.test.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+jest.mock("../../util/Routing");
+jest.mock("../../util/Ajax", () => {
+    const originalModule = jest.requireActual("../../util/Ajax");
+    return {
+        ...originalModule,
+        default: jest.fn()
+    };
+});
+
+const COMMENT_TYPE = "http://rdfs.org/sioc/types#Comment";
```
Comment (psiotwo) [inline src/action/__tests__/AsyncCommentActions.test.ts]: Into VocabularyUtils?

File: src/component/comment/CommentDislikes.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import withI18n, {HasI18n} from "../hoc/withI18n";
+import {injectIntl} from "react-intl";
+import {isMine} from "./CommentLikes";
+
+interface CommentDislikesProps extends HasI18n {
+    reactions: CommentReaction[];
+    comment: Comment;
+
+    currentUser: User;
+    addReaction: (comment: Comment, reactionType: string) => void;
```
Comment (psiotwo) [inline src/component/comment/CommentDislikes.tsx:16]: maybe some type for reactionTypes - like/dislike ?

File: src/component/comment/Comments.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    removeReaction: (comment: Comment) => Promise<any>;
+}
+
+const Comments: React.FC<CommentsProps> = props => {
+    const {loadComments, createComment, addReaction, removeReaction, updateComment, term} = props;
+    const [comments, setComments] = React.useState<Comment[]>([]);
+    React.useEffect(() => {
+        loadComments(term.iri).then(data => setComments(data));
+    }, [loadComments, term]);
+    const onSubmit = (comment: Comment) => {
```
Comment (psiotwo) [inline src/component/comment/Comments.tsx]: "{ return" redundant?

File: src/i18n/en.ts [hunk 4]
```diff
@@ -549,6 +549,16 @@ const en = {
         "public.dashboard.actions.login": "Log in",
         "public.dashboard.actions.register": "Create account",
         "public.dashboard.actions.vocabularies": "Browse vocabularies and terms",
+
+        "comments.title": "Comments",
+        "comments.create.placeholder": "Type your comment...",
+        "comments.create.submit.title": "Submit",
+        "comments.list.empty": "No comments, yet.",
+        "comments.comment.like": "Like this comment",
+        "comments.comment.like.on": "Cancel reaction",
```
Comment (psiotwo) [inline src/i18n/en.ts]: Misaligned with the czech version:
"comments.comment.like.on": "Už se mi nelíbí",
"comments.comment.like.on": "Cancel reaction"

Maybe change the Czech version to "Zrušit reakci", until we come up with some consistent naming for the actions (with meanings - like / cancel like / dislike / cancel dislike - but better, more understandable)

**Review by psiotwo — COMMENTED** [review]
see comments

File: src/component/comment/Comments.tsx (hunk 3 above)
Comment (ledsoft) [reply src/component/comment/Comments.tsx]: Was going for readability, but ok.

File: src/component/comment/CommentDislikes.tsx (hunk 2 above)
Comment (ledsoft) [reply src/component/comment/CommentDislikes.tsx:16]: Added constant with literal values. Couldn't use enum, asi t does not allow computed values (NS + Like/Dislike)

File: config/server.json [hunk 5]
```diff
@@ -1,3 +1,3 @@
 {
-  "url": "TO-BE-PROVIDED"
+  "url": "http://localhost:8081/termit"
```
Comment (psiotwo) [inline config/server.json]: leaked URL

**Review by psiotwo — CHANGES_REQUESTED** [review]

**Review by psiotwo — APPROVED** [review]
lgtm

**Pull request merged**